xarray/core/utils.py

```diff
@@ -96,10 +96,16 @@ def equivalent(first, second):
     """Compare two objects for equivalence (identity or equality), using
     array_equiv if either object is an ndarray
     """
+    def both_nan(first, second):
+        try:
+            return np.isnan(first) and np.isnan(second)
```

Comment:
Use `pd.isnull` instead of `np.isnan`. That should be more robust and avoid the need for the try/except clause.